### PR TITLE
[WFLY-19190] Update the jaxrs subsystem resource descriptions to be more useful.

### PR DESCRIPTION
**Pull request by soul2zimate**
issue: https://issues.redhat.com/browse/WFLY-19190

[WFLY-19190] Update the jaxrs subsystem resource descriptions to be more useful.

____

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties [hunk 1]
```diff
@@ -35,31 +35,31 @@ jaxrs.deployment.sub-resource-locators.produces=Produce types of the Jakarta RES
...
+jaxrs.resteasy-language-mappings=resteasy.language.mappings configuration parameter. Replaces the need for an Accept-Language header by mapping file name extensions (like .en or .fr) to a language. Used when the client is unable to use an Accept-Language header to choose a language (i.e. a browser).
+jaxrs.resteasy-media-type-mappings=resteasy.media.type.mappings configuration parameter. Replaces the need for an Accept header by mapping file name extensions (like .xml or .txt) to a media type. Used when the client is unable to use an Accept header to choose a representation (i.e. a browser).
+jaxrs.resteasy-media-type-param-mapping=resteasy.media.type.param.mapping configuration parameter. Names a query parameter that can be set to an acceptable media type, enabling content negotiation without an Accept header.
+jaxrs.resteasy-prefer-jackson-over-jsonb=resteasy.prefer-jackson-over-jsonb configuration parameter.  Specifies that the Jackson JSON provider should be preferred over the default Jettison JSON provider for JSON-binding.
```
Comment (jamezp) [inline jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties:55]: This should be:
> Specifies that the Jackson JSON provider should be preferred over the default Jakarta JSON Processing provider for JSON binding.

**Review by jamezp — CHANGES_REQUESTED** [review]
I think we should remove the property prefix too, e.g. 
> resteasy.add.charset configuration parameter.

I don't see a need for keeping the property name there.

File: jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties [hunk 2]
```diff
@@ -35,31 +35,31 @@ jaxrs.deployment.sub-resource-locators.produces=Produce types of the Jakarta RES
 jaxrs.deployment.sub-resource-locators.java-method=Java method of the Jakarta RESTful Web Services resource.
 jaxrs.deployment.sub-resource-locators.resource-methods=Annotated methods for the Jakarta RESTful Web Services resource.
 
-jaxrs.jaxrs-2-0-request-matching=jaxrs.2.0.request.matching configuration parameter
-jaxrs.resteasy-add-charset=resteasy.add.charset configuration parameter
-jaxrs.resteasy-buffer-exception-entity=resteasy.buffer.exception.entity configuration parameter
-jaxrs.resteasy-disable-html-sanitizer=resteasy.disable.html.sanitizer configuration parameter
-jaxrs.resteasy-disable-providers=resteasy.disable.providers configuration parameter
-jaxrs.resteasy-document-expand-entity-references=resteasy.document.expand.entity.references configuration parameter
+jaxrs.jaxrs-2-0-request-matching=jaxrs.2.0.request.matching configuration parameter. In searching for a matching resource method with which to respond to a request, consider only resource methods with the best match for the request path.
```
[the remark this answers is not in the document]
Comment (soul2zimate) [reply jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties]: Fixed.

File: jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties (hunk 1 above)
Comment (soul2zimate) [reply jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties]: Fixed.

File: jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties [hunk 3]
```diff
@@ -35,31 +35,31 @@ jaxrs.deployment.sub-resource-locators.produces=Produce types of the Jakarta RES
...
+jaxrs.resteasy-add-charset=resteasy.add.charset configuration parameter. If a resource method returns a text/* or application/xml* media type without an explicit charset, RESTEasy will add "charset=UTF-8" to the returned Content-Type header. Note that the charset defaults to UTF-8 in this case, independent of the setting of this parameter.
+jaxrs.resteasy-buffer-exception-entity=resteasy.buffer.exception.entity configuration parameter. Upon receiving an exception, the client side buffers any response entity before closing the connection.
+jaxrs.resteasy-disable-html-sanitizer=resteasy.disable.html.sanitizer configuration parameter. Normally, a response with media type "text/html" and a status of 400 will be processed so that the characters "/", "<", ">", "&", """ (double quote), and "'" (single quote) are escaped to prevent an XSS attack. Setting this parameter to "true", escaping will not occur.
+jaxrs.resteasy-disable-providers=resteasy.disable.providers configuration parameter. A comma delimited list of fully qualified JAX-RS @Provider class names that will be disabled.
```
[the remark this answers is not in the document]
Comment (soul2zimate) [reply jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties]: Fixed

**Comment by soul2zimate** [comment]
Thank you. I have rebased as per your review comments.

**Review by jamezp — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @soul2zimate and @jamezp

**Pull request merged**